lib/charms/grafana_agent_k8s/v0/grafana_agent.py

```diff
@@ -254,18 +307,15 @@ def _build_config_file(self, event) -> str:
 
         return yaml.dump(config)
 
-    def _initial_config(self) -> dict:
+    def _initial_config(self) -> None:
```

Comment:
The name of this method really does not suggest you are pushing stuff.

I'd keep `_initial_config` as a property, and do the pushing somewhere else.